framework/src/systems/NonlinearSystemBase.C

```diff
@@ -1130,9 +1137,18 @@ NonlinearSystemBase::setConstraintSecondaryValues(NumericVector<Number> & soluti
             // reinit variables on the primary element's face at the contact point
             _fe_problem.setNeighborSubdomainID(primary_elem, 0);
             subproblem.reinitNeighborPhys(primary_elem, primary_side, points, 0);
+            if (displaced)
+              _fe_problem.reinitNeighborPhys(
+                  _mesh.elemPtr(primary_elem->id()), primary_side, points, 0);
```

Comment:
So most routines in `FEProblemBase` call to the same routine in `DisplacedProblem` but this is one of the few that doesn't. And that's because physical points which fall on a given element ID on one mesh are not guaranteed to fall on the same element ID in the other mesh. For example if a domain expands and you take points along the boundary of the displaced mesh, then these points won't fall on *any* element in the undisplaced mesh. 

This is why SAM is failing ... libMesh is unable to map the physical points into the reference coordinate space of the undisplaced mesh element because the points don't lie on the element

Reply:
> because the points don't lie on the element

Is the difference between the two calls a _side_ restriction?

Reply:
I'm not sure what you mean. The only way I can think of to make this work is to pass the reference points from the inversion of the physical points in the first `reinitNeighborPhys` call to a second `reinitNeighborXXX` call. We allow something like this for cohesive zone modeling which has the same problem you're running into here